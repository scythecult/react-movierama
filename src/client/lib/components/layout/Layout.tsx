import clsx from 'clsx';
import type { PropsWithChildren } from 'react';
import type { PropsWithClassName } from '../../types/PropsWithClassName';
import styles from './styles.module.css';

export type LayoutProps = PropsWithChildren<PropsWithClassName>;

export const Layout = (props: LayoutProps) => {
  const { className, children } = props;
  const classNameFinal = clsx(styles.layout, className);

  return <div className={classNameFinal}>{children}</div>;
};
