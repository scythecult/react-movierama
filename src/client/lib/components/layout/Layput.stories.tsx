import type { Meta, StoryFn } from '@storybook/react-vite';
import { Layout, type LayoutProps } from './Layout';

export default {
  title: 'Components/Layout',
  component: Layout,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof Layout>;

const Template: StoryFn<LayoutProps> = (props) => <Layout {...props} />;

export const Default = Template.bind(null);
Default.args = {
  children: <div style={{ backgroundColor: 'red' }}>Lorem ipsum</div>,
};
