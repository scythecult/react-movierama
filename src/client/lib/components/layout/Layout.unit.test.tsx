import { render } from '@testing-library/react';
import { Layout } from '@/client/lib/components/layout/Layout';

describe('Layout', () => {
  test('should correspond default layout', () => {
    const result = render(<Layout />);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "children" prop', () => {
    const result = render(<Layout>Children</Layout>);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "className" prop', () => {
    const result = render(<Layout className="custom-class">Children</Layout>);

    expect(result.container).toMatchSnapshot();
  });
});
