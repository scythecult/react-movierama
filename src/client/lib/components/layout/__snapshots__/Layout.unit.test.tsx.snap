// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > should correspond default layout 1`] = `
<div>
  <div
    class="_layout_762df9"
  />
</div>
`;

exports[`Layout > should support the "children" prop 1`] = `
<div>
  <div
    class="_layout_762df9"
  >
    Children
  </div>
</div>
`;

exports[`Layout > should support the "className" prop 1`] = `
<div>
  <div
    class="_layout_762df9 custom-class"
  >
    Children
  </div>
</div>
`;
